tests/test_autodoc.py

```diff
@@ -473,3 +474,135 @@ def test_autodoc_getset_descriptor(self):
 
 </div>\n"""
         self.assertEqual(documentation, expected_documentation)
+
+    def test_hashlink_example_codeblock(self):
+        dummy_anchor = "myfunc"
+        # test canonical
+        original_md = """Example:
+```python
+import numpy as np
+```"""
+        expected_conversion = """<ExampleCodeBlock anchor="myfunc.example">
+
+Example:
+```python
+import numpy as np
+```
+
+</ExampleCodeBlock>"""
+        self.assertEqual(hashlink_example_codeblock(original_md, dummy_anchor), expected_conversion)
+
+        # test `Examples` ending in `s`
+        original_md = """Examples:
+```python
+import numpy as np
+```"""
+        expected_conversion = """<ExampleCodeBlock anchor="myfunc.example">
+
+Examples:
+```python
+import numpy as np
+```
+
+</ExampleCodeBlock>"""
+        self.assertEqual(hashlink_example_codeblock(original_md, dummy_anchor), expected_conversion)
+
+        # test part of bigger doc description
+        original_md = """Some description about this function
+Example:
+```python
+import numpy as np
+```"""
+        expected_conversion = """Some description about this function
+<ExampleCodeBlock anchor="myfunc.example">
+
+Example:
+```python
+import numpy as np
+```
+
+</ExampleCodeBlock>"""
+        self.assertEqual(hashlink_example_codeblock(original_md, dummy_anchor), expected_conversion)
+
+        # test complex example introduction
+        original_md = """Here is a classification example:
+```python
+import numpy as np
+```"""
+        expected_conversion = """<ExampleCodeBlock anchor="myfunc.example">
+
+Here is a classification example:
+```python
+import numpy as np
+```
+
+</ExampleCodeBlock>"""
+        self.assertEqual(hashlink_example_codeblock(original_md, dummy_anchor), expected_conversion)
+
+        # test doc description with multiple examples
+        original_md = """Here is a classification example:
+```python
+import numpy as np
+```
+
+Here is a regression example:
+```python
+import scipy as sp
+```"""
+        expected_conversion = """<ExampleCodeBlock anchor="myfunc.example">
+
+Here is a classification example:
+```python
+import numpy as np
+```
+
+</ExampleCodeBlock>
+
+<ExampleCodeBlock anchor="myfunc.example-2">
+
+Here is a regression example:
+```python
+import scipy as sp
+```
+
+</ExampleCodeBlock>"""
+        self.assertEqual(hashlink_example_codeblock(original_md, dummy_anchor), expected_conversion)
+
+        # test example with inline ``` (inline ``` should be escaped)
+        original_md = """The tokenization method is `<tokens> <eos> <language code>` for source language documents, and ``<language code>
```

Comment:
```suggestion
        original_md = """The tokenization method is `<tokens> <eos> <language code>` for source language documents, and ```<language code>
```
or did I misunderstand what's tested?

Reply:
handled in https://github.com/huggingface/doc-builder/pull/197/commits/f046870c57ffc08d293835f63eca3239453334ee